test/suites/integration/bsp-volunteer.test.ts

```diff
@@ -0,0 +1,125 @@
+
+describe("BSPNet: BSP Volunteer", () => {
+  let api: BspNetApi;
+
+  before(async () => {
+    await runBspNet();
+    api = await createApiObject(`ws://127.0.0.1:${NODE_INFOS.user.port}`);
+  });
+
+  after(async () => {
+    await cleardownTest(api);
+  });
+
+  it("Network launches and can be queried", async () => {
+    const userNodePeerId = await api.rpc.system.localPeerId();
+    strictEqual(userNodePeerId.toString(), NODE_INFOS.user.expectedPeerId);
+
+    const bspApi = await createApiObject(`ws://127.0.0.1:${NODE_INFOS.bsp.port}`);
+    const bspNodePeerId = await bspApi.rpc.system.localPeerId();
+    bspApi.disconnect();
+    strictEqual(bspNodePeerId.toString(), NODE_INFOS.bsp.expectedPeerId);
+  });
+
+  it("file is finger printed correctly", async () => {
+    const source = "res/adolphus.jpg";
+    const destination = "test/adolphus.jpg";
+    const { fingerprint, size, location } = await api.sendFile(
+      source,
+      destination,
+      NODE_INFOS.user.AddressId
+    );
+
+    strictEqual(hexToString(location), destination);
+    strictEqual(fingerprint, TEST_ARTEFACTS[source].fingerprint);
+    strictEqual(size, TEST_ARTEFACTS[source].size);
+  });
+
+  it("issueStorageRequest sent correctly", async () => {
+    const source = "res/smile.jpg";
+    const destination = "test/smile.jpg";
+    const { fingerprint, size, location } = await api.sendFile(
+      source,
+      destination,
+      NODE_INFOS.user.AddressId
+    );
+
+    const result = await api.sealBlock(
+      api.tx.fileSystem.issueStorageRequest(location, fingerprint, size, DUMMY_MSP_ID, [
+        NODE_INFOS.user.expectedPeerId,
+      ]),
+      shUser
+    );
+
+    const { event } = api.assertEvent("fileSystem", "NewStorageRequest", result.events);
+
+    const dataBlob = api.events.fileSystem.NewStorageRequest.is(event) && event.data;
+
+    if (!dataBlob) {
+      throw new Error("Event doesn't match Type");
+    }
+
+    strictEqual(dataBlob.who.toString(), NODE_INFOS.user.AddressId);
+    strictEqual(dataBlob.location.toHuman(), destination);
+    strictEqual(dataBlob.fingerprint.toString(), fingerprint);
+    strictEqual(dataBlob.size_.toBigInt(), size);
+    strictEqual(dataBlob.peerIds.length, 1);
+    strictEqual(dataBlob.peerIds[0].toHuman(), NODE_INFOS.user.expectedPeerId);
+  });
+
+  it("bsp volunteers when issueStorageRequest sent", async () => {
+    const source = "res/whatsup.jpg";
+    const destination = "test/whatsup.jpg";
+    const { fingerprint, size, location } = await api.sendFile(
+      source,
+      destination,
+      NODE_INFOS.user.AddressId
+    );
+
+    await api.sealBlock(
+      api.tx.fileSystem.issueStorageRequest(location, fingerprint, size, DUMMY_MSP_ID, [
+        NODE_INFOS.user.expectedPeerId,
+      ]),
+      shUser
+    );
+
+    const pending = await api.rpc.author.pendingExtrinsics();
+    strictEqual(pending.length, 1, "There should be one pending extrinsic from BSP");
```

Comment:
In the first execution of this test, it worked like a charm. However after executing it for a second, third and so on, I'm always getting two `bspVolunteer` extrinsics here for some reason. I attach the output (I added some console logs to print the methods of the two extrinsics):

```
facundofarall@Facundos-MacBook-Pro test % pnpm test:bspnet

> @storagehub/test@0.1.0 test:bspnet /Users/facundofarall/Desktop/Moonsong/storage-hub/test
> node --no-deprecation --import tsx --test ./suites/integration/**.test.ts

sh user id: 5CombC1j5ZmdNMEpWYpeEWcKPPYcKsC1WgMPgzGLU72SLa4o
sh bsp id: 5FHSHEFWHVGDnyiw66DoRUpLyh5RouWkXo9GT1Sjk8qw7MAg
 Network docker_default  Creating
 Network docker_default  Created
 Container docker-sh-bsp-1  Creating
 Container docker-sh-bsp-1  Created
 Container docker-sh-bsp-1  Starting
 Container docker-sh-bsp-1  Started
sh-bsp IP: 172.30.0.2
Waiting for node at http://127.0.0.1:9966 to launch...
Waiting for node at http://127.0.0.1:9966 to launch...
Waiting for node at http://127.0.0.1:9966 to launch...
Waiting for node at http://127.0.0.1:9966 to launch...
Waiting for node at http://127.0.0.1:9966 to launch...
Waiting for node at http://127.0.0.1:9966 to launch...
Waiting for node at http://127.0.0.1:9966 to launch...
sh-bsp Peer ID: 12D3KooWNEZ8PGNydcdXTYy1SPHvkP9mbxdtTqGGFVrhorDzeTfU
 Container docker-sh-user-1  Creating
 Container docker-sh-user-1  Created
 Container docker-sh-user-1  Starting
 Container docker-sh-user-1  Started
sh-user Peer ID: 12D3KooWMvbhtYjbhgjoDzbnf71SFznJAKBBkSGYEUtnpES1y9tM
▶ BSPNet: BSP Volunteer
  ✔ Network launches and can be queried (211.077541ms)
  ✔ file is finger printed correctly (22.709375ms)
  ✔ issueStorageRequest sent correctly (53.449541ms)
First pending extrinsic bspVolunteer
Second pending extrinsic bspVolunteer
  ✖ bsp volunteers when issueStorageRequest sent (22.983ms)
    AssertionError [ERR_ASSERTION]: There should be one pending extrinsic from BSP
        at TestContext.<anonymous> (/Users/facundofarall/Desktop/Moonsong/storage-hub/test/suites/integration/bsp-volunteer.test.ts:106:5)
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async Test.run (node:internal/test_runner/test:640:9)
        at async Suite.processPendingSubtests (node:internal/test_runner/test:382:7) {
      generatedMessage: false,
      code: 'ERR_ASSERTION',
      actual: 2,
      expected: 1,
      operator: 'strictEqual'
    }

 Container docker-sh-user-1  Stopping
 Container docker-sh-bsp-1  Stopping
 Container docker-sh-bsp-1  Stopped
 Container docker-sh-bsp-1  Removing
 Container docker-sh-bsp-1  Removed
 Container docker-sh-user-1  Stopped
 Container docker-sh-user-1  Removing
 Container docker-sh-user-1  Removed
 Network docker_default  Removing
 Network docker_default  Removed
▶ BSPNet: BSP Volunteer (7632.693542ms)

ℹ tests 4
ℹ suites 1
ℹ pass 3
ℹ fail 1
ℹ cancelled 0
ℹ skipped 0
ℹ todo 0
ℹ duration_ms 13330.519666

✖ failing tests:

test at file:/Users/facundofarall/Desktop/Moonsong/storage-hub/test/suites/integration/bsp-volunteer.test.ts:1:2191
✖ bsp volunteers when issueStorageRequest sent (22.983ms)
  AssertionError [ERR_ASSERTION]: There should be one pending extrinsic from BSP
      at TestContext.<anonymous> (/Users/facundofarall/Desktop/Moonsong/storage-hub/test/suites/integration/bsp-volunteer.test.ts:106:5)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Test.run (node:internal/test_runner/test:640:9)
      at async Suite.processPendingSubtests (node:internal/test_runner/test:382:7) {
    generatedMessage: false,
    code: 'ERR_ASSERTION',
    actual: 2,
    expected: 1,
    operator: 'strictEqual'
  }
 ELIFECYCLE  Command failed with exit code 1.
```

Reply:
This is disconcerting, let's see if we can repro with latest version of scripts

Reply:
This was fixed by adding a static wait to allow time for the BSP to volunteer as expected.